### DISCUSSION TITLE
Preload image in scss with hash generated

### DISCUSSION THREAD

**Discussion by mad4hit** [nuxt/nuxt#23330] [Questions] [answered]
### Environment

Nuxt project info:                                                                                                                12:25:29

------------------------------
- Operating System: Windows_NT
- Node Version:     v18.16.1
- Nuxt Version:     3.7.1
- CLI Version:      3.7.3
- Nitro Version:    2.6.2
- Package Manager:  npm@9.8.1
- Builder:          -
- User Config:      runtimeConfig, modules, app, experimental, css, typescript, alias, multiCache, nitro, vite
- Runtime Modules:  @nuxt/devtools@0.8.3, nuxt-lazy-hydrate@1.0.0, nuxt-lodash@2.5.0, nuxt-multi-cache@3.0.1
- Build Modules:    -
------------------------------

### Reproduction

Hi, 
my goal is to preload the hero image in the homepage using meta link preload  and i have this configuration:

**in nuxt.config.js:**

`export default defineNuxtConfig({
 app: {
  head: {
 link: [
 {
          rel: "preload",
          href: "_nuxt/back-hero-image.webp",
          as: "image",
        }
]
}
}
}
`

**in scss i have:**

```
 .hero-image {
    background-image: url("../assets/back-hero-image.webp");

  }
```

 

When the app is builded the image in the asset is hashed and the configuration in nuxtConfig fails to generate meta link because it generate  

`<link rel="preload" href="_nuxt/back-hero-image.webp" as="image">`

but in the css generated i got 
` 
.hero-image {
background-image:url(/_nuxt/back-hero-image.1dd51c09.webp))
}
`

### Describe the bug

The image is downloaded twice, one in meta preload and the other when is requider for the background with the hash.

**Comment by logotip4ik** [comment, marked as answer]
Hello! To preload image from assets folder you will need to leverage vite to get correct link and then use `useHead` composable to create preload link to that image.

To get image link you need to prefix import path with `?url` to get something like this: `import ImageUrl from '~/assets/image.jpg?url'`

Then create preload link with `useHead`:

```html
<script setup>
import ImageUrl from '~/assets/image.jpg?url'

useHead({
  link: [
    { rel: 'preload', as: 'image', href: ImageUrl },
  ]
})
</script>
```

Here is working demo: https://codesandbox.io/p/sandbox/affectionate-yalow-zd96vp?file=/app.vue:5,59

**Reply by mad4hit** [reply]
Hi @logotip4ik ,

thanks for the explanation.

Your solution works!
